### PR TITLE
[MM-47187] Certificate auth for Elasticsearch

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
Adds support for client certificates and a CA to the Elasticsearch config.

#### Ticket Link
[MM-47187](https://mattermost.atlassian.net/browse/MM-47187)

#### Release Note
```release-note
NONE
```

[MM-47187]: https://mattermost.atlassian.net/browse/MM-47187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

File: services/telemetry/telemetry.go [hunk 1]
```diff
@@ -777,6 +777,9 @@ func (ts *TelemetryService) trackConfig() {
 		"bulk_indexing_batch_size": *cfg.ElasticsearchSettings.BatchSize,
 		"request_timeout_seconds":  *cfg.ElasticsearchSettings.RequestTimeoutSeconds,
 		"skip_tls_verification":    *cfg.ElasticsearchSettings.SkipTLSVerification,
+		"ca":                       *cfg.ElasticsearchSettings.CA,
+		"client_cert":              *cfg.ElasticsearchSettings.ClientCert,
+		"client_key":               *cfg.ElasticsearchSettings.ClientKey,
```
Comment (isacikgoz) [inline services/telemetry/telemetry.go:782]: Let's use `isDefault()` for this, i don't think otherwise it will bring much value.

**Review by isacikgoz — COMMENTED** [review]
Looks good, just a small note on telemetry.

File: services/telemetry/telemetry.go (hunk 1 above)
Comment (noxer) [reply services/telemetry/telemetry.go]: Good point. Changed it.

**Review by isacikgoz — APPROVED** [review]
LGTM!

**Pull request merged**